site/content/Reference/_index.md

```diff
@@ -0,0 +1,248 @@
+
+## Specifying Inputs and Outputs
+The core command line flags for `vhs` are focussed on defining the data flow that `vhs` will use for a recording/replay
+session. Inputs and outputs are specified in terms of a simple domain specific language that will be detailed in
+the following sections.
+
+### Inputs
+```--input "<source|modifier(s)|format>"```
+
+Inputs are specified in a pipe-delimited (`|`), double-quoted string following the `--input` command line flag. 
+The first element in the string must specify a [source](#sources), followed by a pipe character, followed by zero or
+more [modifiers](#input-modifiers) separated by pipe characters, followed by another pipe character, and ending with an
+input [format](#input-formats) specifier. The specified source originates a data stream that is modified by the 
+specified modifiers and then formatted, or interpreted by the specified input format.
+
+In the example command given above, the input specifier is `--inputs "tcp|http"` where `tcp` specifies the TCP source
+and `http` specifies the HTTP input format. This example does not use any input modifiers. 
+
+The next sections will detail the currently available [input sources](#input-sources), [modifiers](#input-modifiers),
+and [formats](#input-formats) in `vhs`. Each source, modifier, and format may require additional configuration in the
+form of additional command line flags. These will be described where applicable.
+
+Only one input definition can be specified in a `vhs` session.
+
+#### Sources
+The following sources are currently available:
+* `tcp`
+* `file`
+* `gcs` (Google cloud storage)
+* `s3compat` (S3 compatible cloud storage)
+
+##### `tcp`
+The `tcp` source captures live TCP/IP network data. It uses the following additional command line flags for
+configuration:
+* `--address <ip address:port>` Required. Specifies the address and port on which `vhs` will listen.
+* `--capture-response` Optional. If set, `vhs` captures requests and responses (2-way traffic).
+
+##### `file`
+The `file` source reads data from a file on the local filesystem. It requires the following command line flag
+for configuration.
+* `--input_file <path to input file>` Required. Specifies the path to the input file to be read.
+
+##### `gcs`
+The `gcs` source reads data from a Google Cloud Storage object. It requires the following command line flags for
+configuration. Note that the GCS source also requires Google Cloud authentication credentials to be present
+on the machine or in the container where `vhs` is run. For more information on GCS authentication, see Google's
+documentation [here](https://cloud.google.com/docs/authentication/production).
+* `--gcs-bucket-name <GCS bucket name>` Required. Name of bucket that contains the object to be read.
+* `--gcs-object-name <object name>` Required. Name of object to be read.
+
+Note that this source also requires a JSON key file containing Google Cloud authentication credentials.
```

Comment:
Might be worth mentioning the use of `GOOGLE_APPLICATION_CREDENTIALS` to specify the location of the credentials.

Reply:
👍

Reply:
Added.